src/core/Akka.Streams/Implementation/Sources.cs

```diff
@@ -761,6 +762,259 @@ public UnfoldResourceSourceAsync(Func<Task<TSource>> create, Func<TSource, Task<
         public override string ToString() => "UnfoldResourceSourceAsync";
     }
 
+    /// <summary>
+    /// INTERNAL API
+    /// </summary>
+    /// <typeparam name="TOut">TBD</typeparam>
+    /// <typeparam name="TSource">TBD</typeparam>
+    /// <typeparam name="TCreateState">The state passed to resource create function</typeparam>
+    [InternalApi]
+    public sealed class UnfoldResourceSourceValueTaskAsync<TOut, TCreateState, TSource> : GraphStage<SourceShape<TOut>>
+    {
+        #region Logic
+
+        private sealed class Logic : OutGraphStageLogic
+        {
+            private readonly UnfoldResourceSourceValueTaskAsync<TOut, TCreateState, TSource> _stage;
+            private readonly Lazy<Decider> _decider;
+            private Option<TSource> _state = Option<TSource>.None;
+
+            private readonly PooledValueTaskContinuationHelper<Option<TOut>>
+                _pooledContinuation;
```

Comment:
It's entirely possible that this entire pattern is overkill and we should consider using `Result` or `Result`+`Holder` patterns consistently throughout these implementations for consistency's sake.

Reply:
or, maybe, the 'hold local' pattern we use for the `UnfoldValueTaskAsync`?

Reply:
I'd stick with one implementation pattern for this - I think the Holder pattern is easier to reason about so far but I haven't had to debug any of this code yet :p

Reply:
Holder pattern is easier to reason about, Especially from a 'safety' standpoint...

But since these _are_ internals, I would expect that unless the `ActorGraphInterpreter` is broken, that the debugging experience would not be overly diminished for maintainers; instead of peeking at `Result` property state they just need to instead look at the `ValueTask` state. Also it's a -little- less code to maintain.

The win, is that directly holding the VT removes indirection `Result` introduces

Note: We do need to keep `Holder` however for `SelectAsync` due to it's internal parallelism and logic rules.

Really though, I want to have benchmarks of `Task` vs this state _first_ before I try to standardize things, _just in case_ the difference of keeping a specific impl is worth a discussion, and/or if whichever we 'pick' shows notable regression (which means, maybe try the other one.) Kinda the other part of why I tried a few different strategies here, This may be important for Artery.